safe-global/web-core
# fix: Stay on the same route when switching safes

## What it solves

Resolves #1734 

Note: Some Safe apps are only available on certain networks. Switching to a safe on a network without that safe app can result in unexpected behaviour.

## How this PR fixes it

- Keeps the same `pathname` when switching safes except when the user is on a safe unspecific route like `/welcome`

## How to test it

1. Navigate to a Safe
2. Switch to a different Safe
3. Observe still being on the same route
4. Navigate to the welcome page
5. Open a Safe
6. Observe being on the Dashboard
7. Navigate to a single transaction
8. Switch to a different Safe
9. Observe the transaction being replaced with a loading screen

## Screenshots

## Checklist
